src/bootstrap/setup.rs

```diff
@@ -1,11 +1,56 @@
 use crate::t;
 use std::path::{Path, PathBuf};
+use std::str::FromStr;
 use std::{
-    env, fs,
+    env, fmt, fs,
     io::{self, Write},
 };
 
-pub fn setup(src_path: &Path, include_name: &str) {
+#[derive(Clone, Copy)]
+pub enum Profile {
+    Compiler,
+    Codegen,
+    Library,
+    User,
+}
+
+impl Profile {
+    fn include_path(&self, src_path: &Path) -> PathBuf {
+        PathBuf::from(format!("{}/src/bootstrap/defaults/config.toml.{}", src_path.display(), self))
+    }
+}
+
+#[derive(Debug)]
+pub struct ProfileErr {
+    pub name: String,
+}
+
+impl FromStr for Profile {
+    type Err = ProfileErr;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "a" | "lib" | "library" => Ok(Profile::Library),
+            "b" | "compiler" => Ok(Profile::Compiler),
+            "c" | "llvm" | "codegen" => Ok(Profile::Codegen),
+            "d" | "maintainer" | "user" => Ok(Profile::User),
+            _ => Err(ProfileErr { name: s.to_string() }),
```

Comment:
I'm not happy with having the `a, b, c, d` letters from the prompt as part of the `FromStr` implementation, maybe having a match like this

```rust
match choice_string {
    "a" => Profile::Library,
    "b" => Profile::Compiler,
    "c" => Profile::Codegen,
    "d" => Profile::User,
    other => match other.parse() {
        Ok(profile) => profile,
        Err(prof_err) => { print error, continue;}
    }
}
```
in the prompting function would be better

Reply:
I don't think it's a big deal either way. I kind of like it the way it is since it means there's only one place the match has to be exhaustive, but you could convince me it would be easier to keep a/b/c in sync if it's next to the prompt.

Reply:
You're right, having only one match to keep exhaustive might be best :)